### PR TITLE
Update NSAccessibility

### PR DESCRIPTION
**Pull request by MaddTheSane**
Use the more up-to-date NSAccessibility protocols instead of the deprecated informal protocol.

### CODE REVIEW

File: Controllers/JVChatRoomBrowser.m [hunk 1]
```diff
@@ -61,21 +61,21 @@ - (void) dealloc {
...
 
-	[roomsTable accessibilitySetOverrideValue:NSLocalizedString(@"Chat rooms", "VoiceOver title for chat rooms table") forAttribute:NSAccessibilityDescriptionAttribute];
+	[roomsTable setAccessibilityMarkerTypeDescription:NSLocalizedString(@"Chat rooms", "VoiceOver title for chat rooms table")];
 
 	theColumn = [roomsTable tableColumnWithIdentifier:@"members"];
 	NSImage *headerImage = [NSImage imageFromPDF:@"person"];
 	headerImage.size = NSMakeSize(10., 10.);
 	[[theColumn headerCell] setImage:headerImage];
-	[[theColumn headerCell] accessibilitySetOverrideValue:NSLocalizedString(@"Participants", "VoiceOver title for number of connected users in chat room browser table") forAttribute:NSAccessibilityDescriptionAttribute];
+	[[theColumn headerCell] setAccessibilityMarkerTypeDescription:NSLocalizedString(@"Participants", "VoiceOver title for number of connected users in chat room browser table")];
```
Comment (zadr) [inline Controllers/JVChatRoomBrowser.m]: Is there a doc or a WWDC vid that explains what an accessibility marker is? It looks like you're using it as a description for the table header (as an alternative to its title), but, I don't know much about the new `NSAccessibility` APIs and would like to be able to learn + verify that this is an expected use for the accessibility description.

(The alternative, accidentally doing the wrong thing— and confusing people using Voiceover— seems bad

File: Controllers/JVChatRoomBrowser.m [hunk 2]
```diff
@@ -61,21 +61,21 @@ - (void) dealloc {
...
 
 	[self tableView:roomsTable didClickTableColumn:[roomsTable tableColumnWithIdentifier:_sortColumn]];
 
 	[roomsTable setDoubleAction:@selector( joinRoom: )];
 
 	[showBrowser setBezelStyle:NSRoundedDisclosureBezelStyle];
 	[showBrowser setButtonType:NSOnOffButton];
-	[[showBrowser cell] accessibilitySetOverrideValue:NSLocalizedString(@"Show chat rooms", "VoiceOver title for disclosure checkbox for chat rooms table in Join room dialog") forAttribute:NSAccessibilityDescriptionAttribute];
+	[[showBrowser cell] setAccessibilityRoleDescription:NSLocalizedString(@"Show chat rooms", "VoiceOver title for disclosure checkbox for chat rooms table in Join room dialog")];
```
Comment (zadr) [inline Controllers/JVChatRoomBrowser.m]: Same question here about a doc/vid that covers accessibility roles (e.g.: why is the role not that of a button, with `accessibilityValueDescription` set?)

File: Controllers/JVChatRoomBrowser.m (hunk 1 above)
Comment (MaddTheSane) [reply Controllers/JVChatRoomBrowser.m]: Looking at the docs, I'm misusing `setAccessibilityMarkerTypeDescription:`. I'll see if I can't find a proper replacement.

File: Controllers/MVConnectionsController.m [hunk 3]
```diff
@@ -1165,7 +1165,7 @@ - (void) tableView:(NSTableView *) view willDisplayCell:(id) cell forTableColumn
 			}
 			if( imageName )
 				[cell setImage:[NSImage imageNamed:imageName]];
-			[cell accessibilitySetOverrideValue:title forAttribute:NSAccessibilityValueDescriptionAttribute];
+			[cell setAccessibilityValueDescription:title];
```
[the remark this answers is not in the document]
Comment (MaddTheSane) [reply Controllers/MVConnectionsController.m]: No, as the original used the `NSAccessibilityValueDescriptionAttribute` key, the equivalent method is `setAccessibilityValueDescription:`

[the remark this answers is not in the document]
Comment (zadr) [reply Controllers/MVConnectionsController.m]: Ah, `NSAccessibility_Value_DescriptionAttribute` vs `NSAccessibilityDescriptionAttribute`, I see 👍

**Pull request merged**